docs/workflows/copy.rst

```diff
@@ -47,47 +47,57 @@ depend on the existence of other content units to be "correct" or meaningful. Fo
 #. A Package Group is a group of RPM packages. If the RPM packages that a Package Group contains
    are not present in the same repository, the Package Group is effectively "broken" and won't be
    possible to install correctly on a client system.
+#. A Module consists of many RPM packages (similar to a Package Group). If the module is added to
+   a repository while the packages that its RPMs depend on are not, it may not be possible to
+   install the module on a client system.
 #. A Module can depend on other modules. If those modules are not present in the RPM repo, the
    module will not be installable on a client system.
 #. RPM Packages typically depend on other RPM packages. If a lone RPM package is added to a
    repository without its dependencies, it will potentially not be installable on a client system.
-#. A Module consists of many RPM packages (similar to a Package Group). If the module is added to
-   a repository while the packages that its RPMs depend on are not, it may not be possible to
-   install the module on a client system.
 
-The advanced copy API exists primarily to address these use cases. Unlike the repository
-modification API, when a copy is performed using the RPM copy API it is permitted to additionally
-copy content in the background which you, the user, did not explicitly tell it to copy. For
-example:
+The advanced copy API exists primarily to address these use cases.
+
+In contrast to the repository modification API, when a copy is performed using the RPM copy
+API it is possible for content in addition to that explicitly requested to be copied to the
+destination.  For example:
 
 * When copying an Advisory (Errata) from one repository to another, all of the RPM packages
   directly referenced by the Advisory will also be copied. (note: this is best-effort, if the
   source repository doesn't contain some of the packages referenced, the copy will proceed
   without them).
-* When copying RPM packages from one repository to another, all of the RPM packages that those
-  packages depend on will also be copied.
+* When copying RPM packages from one repository to another, if dependency-resolution is enabled
+  then all of the RPM packages that those packages depend on are also be copied.
+
+.. note::
 
-The goal is to be as easy to use as possible while maintaining the "correctness" of the repository.
+    In all such cases, Pulp3 engages in a "best effort" attempt to fulfill the requirement. If
+    the source repositor(ies) don't contain all of the referenced content, the copy will proceed
+    without them.
```

Comment:
Maybe we should mention that we log it as a warning, and to maybe keep an eye on the logs.

Reply:
To be entirely honest I don't like the fact that we default to best-effort.  I think it should be a non-default config option and while Katello can set it everywhere if they want to, we should be more strict about trying to enforce correctness.

Reply:
I certainly think there's a discussion to have around that, but right now I'm just trying to make the docs clear on what we do Right Now. Revisiting this last section to include your suggestions and make the limitations more clear as well.